charts/k8s-monitoring/templates/alloy_config/_profiles_java.alloy.txt

```diff
@@ -30,6 +30,18 @@ discovery.relabel "java_pods" {
     action = "keep"
     regex = ".*/java$"
   }
+  rule {
+    source_labels = ["__meta_kubernetes_pod_name"]
+    regex = ".+"
+    action = "keep"
+  }
+{{- if .Values.profiles.java.namespaces }}
+  rule {
+    source_labels = ["__meta_kubernetes_namespace"]
+    regex = "({{ join "|" .Values.profiles.java.namespaces }})"
+    action = "keep"
+  }
+{{- end }}
```

Comment:
I don't think it does what you want it to achieve:

This will keep everything that was kept before (java processes) and additionally all pods and additionally the namespace if supplied.

Let me have a think and I propose something what I think will work instead.

Reply:
I am fairly certain the filter for namespaces is already taken care of, but it is nothing I have tested. We might need to deploy to be 100% certain:

The only change I suggest is adding this before the keep role in L28:

```
  rule {
    source_labels = ["__meta_kubernetes_pod_name"]
    regex = "^$"
    action = "drop"
  }
```

Reply:
Thanks! Made those changes!